### ISSUE TITLE
How do use Parameters with web workers 

### ISSUE THREAD

**Issue by cardenasivan** [kripken/sql.js#163]
And do they have to be named?

  sqljsWorker.postMessage({
                id: stringUtils.createGuidCompact(),
                action: workerActions.EXEC,
                sql: sqlQuery
            });

**Comment by lovasoa** [comment]
If you want to use query parameters, then you have to use the action `each`, and specify an additional `params` parameter. ([sql.js source code](https://github.com/kripken/sql.js/blob/master/coffee/worker.coffee#L32)).

You should use something like:
```js
var id = 'REQID';

sqljsWorker.postMessage({
  id: id,
  action: 'each',
  sql: 'SELECT * FROM hello WHERE a=:aval AND b=:bval',
  params: {':aval' : 1, ':bval' : 'world'}
});

sqljsWorker.addEventListener('message', function(e) {
  var msg = e.data;
  if (msg.id === id) {
    if (msg.finished === false) {
      var retrievedRow = msg.row;
      // do something with the row you retrieved from the db
    } else {
      // eventually do something once the last row has been received
    }
  }
});
```

**Comment by cardenasivan** [comment]
Do parameters have to be named or can we do something like 
sqljsWorker.postMessage({
  id: id,
  action: 'each',
  sql: 'SELECT * FROM hello WHERE a= ? AND b=?',
  params: [1, 'world']
});

**Comment by lovasoa** [comment]
Yes, it should work. `db.each()` is called in the worker, so you can use the same argument types as you would with [`db.each(..)`](http://kripken.github.io/sql.js/documentation/class/Database.html#each-dynamic).
